components/ProfileForm/editor.js

```diff
@@ -56,33 +60,108 @@ export const ProfileEditor = ({ profile, setEditMode, mutateProfile }) => {
 
   const handleCancelClick = () => {
     setEditMode(false);
-  };
+  }
+
+  const timezonesEntries = useMemo(() => {
+    return Object.entries(allTimezones);
+  }, [allTimezones])
+  
+  const currentTimezone = spacetime.now().timezone().name;
+  const [currentArea, currentCity] = currentTimezone.split('/');
+  // if currentTimezone === "Europe/Berlin" then select
+  // ['Europe/Amsterdam', 'Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna']
+  // and returns the index [0] -> 'Europe/Amsterdam'
+  const timezonesDefaultValue = useMemo(() => {
+    const timezone = timezonesEntries.filter(
+      ([key, value]) => {
+        const area = key.split('/')[0]; // 'Europe/Amsterdam'
+        const cities = value.split(', '); // 'Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna'
+        return currentArea === area && cities.includes(currentCity);
+      }); // [['Europe/Amsterdam']]
+    return (timezone.length > 0) ? timezone[0][0] : null;
+  }, [timezonesEntries, currentArea, currentCity]);
+
+  const timezonesOptions = useMemo(() => {
+    return timezonesEntries
+      .reduce((selectOptions, zone) => {
+        const now = spacetime.now(zone[0])
+        const tz = now.timezone()
+        const tzStrings = soft(zone[0])
+
+        let abbr = now.isDST() 
+          ? tzStrings[0].daylight?.abbr : tzStrings[0].standard?.abbr
+        let altName = now.isDST()
+          ? tzStrings[0].daylight?.name
+          : tzStrings[0].standard?.name
+
+        const min = tz.current.offset * 60
+        const hr =
+          `${(min / 60) ^ 0}:` + (min % 60 === 0 ? '00' : Math.abs(min % 60))
+        const label = `(GMT${hr.includes('-') ? hr : `+${hr}`}) ${zone[1]}`
+        selectOptions.push({
+          value: tz.name,
+          label: label,
+          offset: tz.current.offset,
+          abbrev: abbr,
+          altName: altName,
+        })
+
+        return selectOptions
+      }, [])
+      .sort((a, b) => a.offset - b.offset)
+  }, [allTimezones])
```

Comment:
definitively! I copied the code as it is from that library and had the same thought. I agree on the idea of saving directly the output of this function run once, instead of using `timezone-list.js` and re-run the function each time. Didn't want to take too much time into this. Thanks for this comment, because I was wondering "is it this how in web I'm supposed to do things?" ^^

Reply:
One small request then: can you add a comment at the top of this code snippet that says it was taken from `https://github.com/ndom91/react-timezone-select`? The license is MIT so it's perfectly fine to take it, but we need to give attribution.

Reply:
I added the credit now in `timezone-list.js`. Looks really good!

Reply:
thanks! nice job on everything!